File: my_cpp_pkg/src/custom_publisher.cpp
#include "my_robot_interfaces/msg/hardware_status.hpp"
#include "rclcpp/rclcpp.hpp"

class CustomPublisherNode : public rclcpp::Node {
public:
    CustomPublisherNode() : Node("custom_publisher") {
        custom_publisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status", 10);
        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                         std::bind(&CustomPublisherNode::publishMessage, this));
        RCLCPP_INFO(this->get_logger(), "Custom publisher has been started.");
    }

    void publishMessage() {
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.temperature = 35.0;
        custom_publisher_->publish(msg);
    }

private:
    rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr custom_publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<CustomPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/custom_publisher.py
#!usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus


class CustomPublisher(Node):
    def __init__(self):
        super().__init__('custom_publisher')
        self.custom_publisher_ = self.create_publisher(
            msg_type=HardwareStatus, topic='hardware_status', qos_profile=10)
        self.timer_ = self.create_timer(
            timer_period_sec=1.0, callback=self.publish_message)
        self.get_logger().info('Custom publisher has been started')

    def publish_message(self):
        msg = HardwareStatus()
        msg.temperature = 34.5
        self.custom_publisher_.publish(msg=msg)
        # self.get_logger().info(f'Number: {self.number_}')


def main(args=None):
    rclpy.init(args=args)
    node = CustomPublisher()
    rclpy.spin(node=node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()
